### PR TITLE
Bug proof: mock mailer doesn't work with recreateApplication:true

### PR DESCRIPTION
**Pull request by Slamdunk**
When the flag `recreateApplication` is set to `true`, the [mockMailer](https://github.com/Codeception/Codeception/blob/2.4.4/src/Codeception/Lib/Connector/Yii2.php#L274) instance is destroyed and recreated too.

If an app page redirects to another page after sending an email, it is impossible to track the email sent.

MockMailer should be destroyed and recreated not at every request, but at every **step**.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,6 +13,9 @@
   "require-dev": {
      "codeception/codeception": "2.4.x-dev"
   },
+  "conflicts": {
+    "codeception/stub": "2.0.1"
```
Comment (Slamdunk) [inline composer.json]: This is due to https://github.com/Codeception/Codeception/issues/5092
I'll drop this as soon as a fix is pushed

**Comment by SamMousa** [comment]
@Slamdunk Thanks for the PR, I'll have a look at this in about 10 days!

File: composer.json (hunk 1 above)
Comment (SamMousa) [reply composer.json]: @Slamdunk please drop it :)

**Review by SamMousa — COMMENTED** [review]
Only thing that needs cleanup is the `composer.json`.

**Comment by Slamdunk** [comment]
Done

**Comment by Slamdunk** [comment]
@SamMousa https://github.com/Codeception/Codeception/pull/5185 got merged, so this should be merged too, shouldn't it?

**Pull request merged**